truss/templates/control/control/helpers/inference_server_process_controller.py

```diff
@@ -47,3 +49,10 @@ def is_inference_server_running(self) -> bool:
             return False
 
         return self._inference_server_process.poll() is None
+
+    def check_and_recover_inference_server(self):
+        if self.inference_server_started() and not self.is_inference_server_running():
+            self._app_logger.warning(
+                "Inference server seems to have crashed, restarting"
```

Comment:
do these logs make it to the user? Is there anything more that we can add that would help them debug the issue?

Reply:
Yes, I think all these logs make it to logs UI. Unlike pynode, where logs are shared across applications and thus we need to be able to tag logs to application, things are simpler here and we can just show all logs to user and we do.

I'm not sure what extra information we can log here, the few crashes I've seen don't leave any information.